dspace-api/src/main/java/org/dspace/content/ItemServiceImpl.java

```diff
@@ -1257,4 +1271,246 @@ public boolean canCreateNewVersion(Context context, Item item) throws SQLExcepti
+                RelationshipMetadataValue metadataValue = constructMetadataValue(context, key);
+                if (metadataValue != null) {
+                    metadataValue = constructResultingMetadataValue(item, value, metadataValue, relationshipId);
+                    metadataValue.setUseForPlace(virtualBean.getUseForPlace());
+                    metadataValue.setPlace(place);
+                    if (StringUtils.isNotBlank(metadataValue.getValue())) {
+                        resultingMetadataValueList.add(metadataValue);
+                    }
+                }
+            }
+        }
+        return resultingMetadataValueList;
+    }
+
+    private RelationshipMetadataValue getRelationMetadataFromOtherItem(Context context, Item otherItem,
+                                                                       String relationName,
+                                                                       Integer relationshipId) {
+        RelationshipMetadataValue metadataValue = constructMetadataValue(context,
+                                                                         MetadataSchemaEnum.RELATION
+                                                                             .getName() + "." + relationName);
+        if (metadataValue != null) {
+            metadataValue.setAuthority(Constants.VIRTUAL_AUTHORITY_PREFIX + relationshipId);
+            metadataValue.setValue(otherItem.getID().toString());
+            return metadataValue;
+        }
+        return null;
+    }
+
+    private String getEntityTypeStringFromMetadata(List<MetadataValue> list) {
+        String entityType = null;
+        for (MetadataValue mdv : list) {
+            if (StringUtils.equals(mdv.getMetadataField().getMetadataSchema().getName(),
+                                   "relationship")
+                && StringUtils.equals(mdv.getMetadataField().getElement(),
+                                      "type")) {
+
+                entityType = mdv.getValue();
+            }
+        }
+        return entityType;
+    }
+
+    private RelationshipMetadataValue constructResultingMetadataValue(Item item, String value,
+                                                                      RelationshipMetadataValue metadataValue,
+                                                                      Integer relationshipId) {
+        metadataValue.setValue(value);
+        metadataValue.setAuthority(Constants.VIRTUAL_AUTHORITY_PREFIX + relationshipId);
+        metadataValue.setConfidence(-1);
+        metadataValue.setDSpaceObject(item);
+        return metadataValue;
+    }
+
+    //This method will construct a RelationshipMetadataValue object with proper schema, element and qualifier based
+    //on the key String parameter passed along to it
+    private RelationshipMetadataValue constructMetadataValue(Context context, String key) {
+        String[] splittedKey = key.split("\\.");
```

Comment:
There is no utility method available. I recently suggests in slack, that we maybe need an object to store metadata field names. The current MetadataField class is a hibernate object and cannot be used to simply store a field. There are already various locations in the (existing) code splitting a string at the dots.

In my opinion there should be a utility class taking a metadata field string (the one with dots), splitting the string and storing the single parts. It would be ideal if this utility class could be used as parameter of the getMetadata() methods. But I do not think, that this should be part of this pull request.

Reply:
https://jira.duraspace.org/browse/DS-4233